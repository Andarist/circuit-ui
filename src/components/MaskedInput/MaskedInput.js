import React from 'react';
import TextMaskInput from 'react-text-mask';

import Input from '../Input';

/**
 * An input supporting text masks to format
 * the input value. Refer to the
 * [README](https://github.com/text-mask/text-mask/tree/master/react#readme)
 * of text-mask for details regarding usage.
 */
const MaskedInput = props => (
  <TextMaskInput
    {...props}
    render={(ref, renderProps) => <Input {...renderProps} deepRef={ref} />}
  />
);

/**
 * @component
 */
export default MaskedInput;
