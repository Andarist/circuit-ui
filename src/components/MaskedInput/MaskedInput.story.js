import React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';

import withTests from '../../util/withTests';
import MaskedInput from './MaskedInput';

storiesOf('MaskedInput', module)
  .addDecorator(withTests('MaskedInput'))
  .add('Default MaskedInput', withInfo()(() => <MaskedInput />));
