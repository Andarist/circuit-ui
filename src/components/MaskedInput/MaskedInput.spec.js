import React from 'react';

import MaskedInput from '.';

describe('MaskedInput', () => {
  /**
   * Style tests.
   */
  it.skip('should render with default styles', () => {
    const actual = create(<MaskedInput />);
    expect(actual).toMatchSnapshot();
  });

  /**
   * Accessibility tests.
   */
  it.skip('should meet accessibility guidelines', async () => {
    const wrapper = renderToHtml(<MaskedInput />);
    const actual = await axe(wrapper);
    expect(actual).toHaveNoViolations();
  });
});
