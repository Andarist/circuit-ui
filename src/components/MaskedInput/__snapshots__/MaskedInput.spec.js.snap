// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MaskedInput should render with default styles 1`] = `
<element
  className="circuit-0 circuit-1"
/>
`;
